### ISSUE TITLE
proper ctrl+c cleanup

### ISSUE THREAD

**Issue by trevorcampbell** [trevorcampbell/fwirl#24]
Right now the main loop doesn't handle Ctrl+C properly. You have to hit Ctrl+C twice to kill the program, at which point it prints a stack trace because the event loop was "already closed".

It boils down to this minimal example. It seems the `to_thread` call is the source of the issue (I am not 100% sure, but I think python is waiting to join that thread but it never ends).
```
import asyncio
from queue import Queue
import time
from threading import Thread


class TestClass:
    def __init__(self):
        self.q = Queue()

    def getter(self):
        val = self.q.get()

    async def _run(self):
        while True:
            task = asyncio.create_task(asyncio.to_thread(self.getter))
            await asyncio.wait([task], return_when=asyncio.FIRST_COMPLETED)  

    def run(self):
        try:
            logger.info('blah')
            asyncio.run(self._run())
        except KeyboardInterrupt:
            sys.stdout.write('hah')
            sys.stdout.flush()

    
t = TestClass()
t.run()
```

**Comment by trevorcampbell** [comment]
One solution would be in `graph.py:_get_message`
```
    def _get_message(self, timeout):
        try:
            msg = self.message_queue.get(timeout=timeout)
        except Empty:
            return None
        return msg
```

instead of passing the timeout to `message_queue.get`, just do a while loop with a 1 second timeout, and count up to the original timeout. Callers of `_get_message` then see a usual timeout, while internally the thread is waking up every second and should catch the interrupt then.

Edit: tried that solution, didn't work. `_get_message` needs to return fully before the interrupt gets processed...

**Comment by trevorcampbell** [comment]
Using the `signal` package to specify a custom handler seems to work

`signal.signal(signal.SIGINT, handler)`

where
```
def handler(sig, frame):
    do something
```

**Comment by trevorcampbell** [comment]
Fixed in 0fd9143
